lxd/resource_lxd_container.go

```diff
@@ -309,6 +309,9 @@ func resourceLxdContainerExists(d *schema.ResourceData, meta interface{}) (exist
 	exists = false
 
 	ct, err := client.ContainerState(name)
+	if err != nil && err.Error() == "not found" {
+		err = nil
```

Comment:
I just thought, exists is a cleaner place to put it. I think if you define it, terraform will use it, otherwise it goes right to Read and you do the d.SetId thing. I'm cool with that approach if you like it better? I just think the Exists function is also wrong and we'd need the fix in both places.